dac: mcux_gau: fix output-voltage-range mapping on RW61x (0,1,3)

Bille-git opened the pull request:
The devicetree binding `nxp,gau-dac` exposes "small|medium|large" with DT enum indices 0..2,
but the RW61x MCUX SDK enum is sparse: {0,1,3} (kDAC_RangeLarge = 3). The driver previously
passed the DT index straight through, so "large" wrote 2 instead of 3 and the highest range
could not be selected.

This maps the DT indices {0,1,2} to the SDK values {0,1,3}.

Boards/SoC: RW612 (e.g., FRDM-RW612)

Signed-off-by: Alexander Steen Sparre Bille assb@foss.dk

Bille-git committed:
test of error fix for GAU-DAC

Bille-git committed:
dac: mcux_gau: map DT range index to SDK values {0,1,3} on RW61x

Bille-git committed:
Update nxp,gau-dac.yaml

removed old test, accidentally left in

martinjaeger reviewed (changes requested):
Thanks for your PR. Please keep the original formatting with tabs instead of spaces.

decsny commented:
changed assignee to myself because I contributed this driver so I will need to review this when I get the chance

Bille-git committed:
dac: mcux_gau: fix indentation style

Replace space-based indentation with tabs to comply with Zephyr’s coding
style and address review feedback.

No functional changes are introduced.

Signed-off-by: Alexander Steen Sparre Bille assb@foss.dk

Bille-git committed:
dac: mcux_gau: fix indentation style

Fix indentation

No functional changes are introduced.

Signed-off-by: Alexander Steen Sparre Bille assb@foss.dk

Bille-git committed:
dac: mcux_gau: fix indentation style

Replace space-based indentation with tabs to comply with Zephyr’s coding
style and address review feedback.

No functional changes are introduced.

Signed-off-by: Alexander Steen Sparre Bille assb@foss.dk

Bille-git committed:
dac: mcux_gau: fix indentation style

Replace space-based indentation with tabs to comply with Zephyr’s coding
style and address review feedback.

No functional changes are introduced.

Signed-off-by: Alexander Steen Sparre Bille assb@foss.dk

Bille-git committed:
dac: mcux_gau: fix indentation style

Added comment

No functional changes are introduced.

Signed-off-by: Alexander Steen Sparre Bille <assb@foss.dk>
